api/cds.proto

```diff
@@ -417,6 +413,16 @@ message Cluster {
   // For instance, if the metadata is intended for the Router filter, the filter
   // name should be specified as *envoy.router*.
   Metadata metadata = 25;
+
+  enum ClusterProtocolSelection {
+    // Cluster can only operate on one of the possible upstream protocols (HTTP1.1, HTTP2).
+    // If :ref:`http2_protocol_options <envoy_api_field_Cluster.http2_protocol_options>` are
+    // present, HTTP2 will be used, otherwise HTTP1.1 will be used.
+    EXCLUSIVE_AS_CONFIGURED = 0;
```

Comment:
USE_CONFIGURED_PROTOCOL sounds better.

Please avoid subtle concept like exclusive unless it is really needed.

Reply:
Done. Thanks for the review!